lib/errorResponseInterceptor.ts

```diff
@@ -22,7 +27,9 @@ export function errorResponseInterceptor(axiosError) {
     axiosError.response.data &&
     // must be a JSON
     // which also implies Content-Type starts with application/json:
-    axiosError.config.responseType === 'json' &&
+    (axiosError.config.responseType === 'json' ||
+      (axiosError.config.responseType === 'stream' &&
+        axiosError[convertedStreamErrorFlag])) &&
```

Comment:
I think now we can use simply:
```suggestion
// must be a JSON
axiosError.response.headers['content-type'].startsWith(
      'application/json'
    )
```

Reply:
I thought about it, but I was a bit worried about other potential `responseType` cases. Apparently the currently allowed values are `'arraybuffer', 'document', 'json', 'text', 'stream'`.
If we toggle only on `application/json` we'd likely break on anything except `json` or `stream` - ofc that is safe for our buuilt-ins, but wasn't sure about any "custom" requests.
Suggest we add this to discussion in https://github.com/IBM/cloudant-node-sdk/pull/1663#pullrequestreview-2413242155 - the flag was only one of the options.